crates/solver/src/settlement_ranker.rs

```diff
@@ -191,17 +184,85 @@ impl SettlementRanker {
+            rated_settlements.retain(|(solver, settlement, _)| {
+                if settlement.score < BigRational::zero() {
+                    tracing::debug!(
+                        solver_name = %solver.name(),
+                        "settlement(s) filtered for having negative score",
+                    );
+                    solver.notify_auction_result(
+                        auction_id,
+                        AuctionResult::Rejected(SolverRejectionReason::NegativeScore(
+                            settlement.score.clone(),
+                        )),
+                    );
+                    return false;
+                }
+                true
+            });
+
+            // Before sorting, make sure to shuffle the settlements. This is to make sure we don't give
+            // preference to any specific solver when there is an objective value tie.
+            rated_settlements.shuffle(&mut rand::thread_rng());
+
+            rated_settlements
+                .sort_by(|a, b| compare_solutions_cip17(&a.1, &b.1, self.decimal_cutoff));
+
+            rated_settlements.iter_mut().rev().enumerate().for_each(
+                |(i, (solver, settlement, _))| {
+                    solver.notify_auction_result(auction_id, AuctionResult::Ranked(i + 1));
+                    self.metrics
+                        .settlement_simulation(solver.name(), SolverSimulationOutcome::Success);
+                    settlement.ranking = i + 1;
+                },
+            );
+        } else {
+            // TODO: remove this block of code once CIP-17 is implemented
+
+            // Before sorting, make sure to shuffle the settlements. This is to make sure we don't give
+            // preference to any specific solver when there is an objective value tie.
+            rated_settlements.shuffle(&mut rand::thread_rng());
+
+            rated_settlements.sort_by(|a, b| compare_solutions(&a.1, &b.1, self.decimal_cutoff));
+
+            let cip17_ranking = cip17_ranking(
+                rated_settlements
+                    .iter()
+                    .map(|(_, settlement, _)| settlement.clone())
+                    .collect(),
+            );
+
+            rated_settlements.iter_mut().rev().enumerate().for_each(
+                |(i, (solver, settlement, _))| {
+                    solver.notify_auction_result(auction_id, AuctionResult::Ranked(i + 1));
+                    self.metrics
+                        .settlement_simulation(solver.name(), SolverSimulationOutcome::Success);
+                    settlement.ranking = cip17_ranking.get(&settlement.id).copied().unwrap_or(0);
+                },
+            );
+        }
         Ok((rated_settlements, errors))
     }
 }
 
+// TODO: remove this once CIP-17 is implemented
+// Sort settlements by CIP-17 rules and return hashmap of settlement id to ranking
+fn cip17_ranking(settlements: Vec<RatedSettlement>) -> HashMap<usize, usize> {
```

Comment:
It's enough to work with references so you don't have to clone all the settlements.

```suggestion
fn cip17_ranking(settlements: Vec<&RatedSettlement>) -> HashMap<usize, usize> {
```

Reply:
Fixed.